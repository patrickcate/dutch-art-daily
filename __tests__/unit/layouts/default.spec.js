import { shallowMount } from '@vue/test-utils'
import DefaultLayout from '@layouts/default.vue'

describe('DefaultLayout Layout', () => {
  it('should be a Vue instance', () => {
    const wrapper = shallowMount(DefaultLayout, {
      stubs: {
        nuxt: '<div></div>',
      },
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
          },
        },
        $route: {
          params: {
            date: '01-01',
          },
        },
      },
    })

    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    const wrapper = shallowMount(DefaultLayout, {
      stubs: {
        nuxt: '<div></div>',
      },
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
          },
        },
        $route: {
          params: {
            date: '01-01',
          },
        },
      },
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('creates root swiper object if undefined', () => {
    const RootComponent = {
      data() {
        return {
          swipers: {
            carousel: true,
          },
        }
      },
    }

    const wrapper = shallowMount(DefaultLayout, {
      parentComponent: RootComponent,
      stubs: {
        nuxt: '<div></div>',
      },
      mocks: {
        $store: {
          state: {
            currentPage: '01-01',
          },
        },
        $route: {
          params: {
            date: '01-01',
          },
        },
      },
    })

    expect(wrapper.vm.$root.swipers).not.toBeUndefined()
    expect(wrapper.vm.$root).toHaveProperty('swipers')
  })

  it('sets active id to current route', () => {
    const mockStore = {
      state: {
        currentPage: null,
      },
      dispatch: jest.fn(),
    }

    shallowMount(DefaultLayout, {
      stubs: {
        nuxt: '<div></div>',
      },
      mocks: {
        $store: mockStore,
        $route: {
          params: {
            date: '01-01',
          },
        },
      },
    })

    expect(mockStore.dispatch).toHaveBeenCalledWith('setCurrentPage', '01-01')
  })
})
