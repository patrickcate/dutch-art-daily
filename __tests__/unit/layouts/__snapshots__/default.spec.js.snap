// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout Layout renders correctly 1`] = `
<div class="l-page">
  <background-stack-stub></background-stack-stub>
  <div class="l-page__content">
    <the-header-stub class="l-page__header"></the-header-stub>
    <div></div>
    <the-footer-stub class="l-page__footer"></the-footer-stub>
  </div>
</div>
`;
